datadog-sidecar/src/service/telemetry/in_proc_receiver.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+// Contains the background task that receives telemetry actions via an internal queue
+// and forwards them to the appropriate telemetry worker.
+
+use crate::service::runtime_info::ActiveApplication;
+use crate::service::telemetry::AppOrQueue;
+use crate::service::{InstanceId, QueueId, SidecarAction, SidecarServer};
+use anyhow::{anyhow, Result};
+use ddcommon::MutexExt;
+use ddtelemetry::worker::TelemetryActions;
+use std::collections::hash_map::Entry;
+use std::sync::OnceLock;
+use tokio::sync::mpsc;
+use tracing::{info, warn};
+
+use super::enqueued_telemetry_data::EnqueuedTelemetryData;
+
+static TELEMETRY_ACTION_SENDER: OnceLock<mpsc::Sender<InternalTelemetryActions>> = OnceLock::new();
```

Comment:
We recently switched from OnceLock to LazyLock, this PR is a bit older, but probably should use the new convention :-)

Reply:
FWICT they serve slightly different purposes: LazyLock requires the initialization code to be provided on construction. In this case, it would require storing in the global variable not just the sending half of the channel, but also the receiving half, since they're created together, which is unnecessary here: only the sending side needs to be accessed from a global.